src/lib/instrument-listen.js

```diff
@@ -51,9 +51,19 @@ module.exports = class InstrumentListen {
 			// Optional fallthrough error handler
 			// eslint-disable-next-line no-unused-vars
 			this.app.use((err, req, res, next) => {
+				let statusCode = err.statusCode || err.status || 500;
+
+				// There's clearly an error, so if the error has a status
+				// code of less than `400` we should default to `500` to
+				// ensure bad error handling doesn't send false positive
+				// status codes
+				if (statusCode < 400) {
+					statusCode = 500;
+				}
+
 				// The error id is attached to `res.sentry` to be returned
 				// and optionally displayed to the user for support.
-				res.statusCode = 500;
+				res.statusCode = statusCode;
```

Comment:
Is any additional validation required? are the values or types for `err.statusCode` and `err.status` enforced anywhere upstream? If not, what are the implications of `res.statusCode` being something unexpected (i.e. not `500`-`599`)?

Reply:
Great question!

Types aren't enforced for either of these properties, they can technically be anything so maybe it does make sense to check this. Digging into Node.js internals [this value does get cast to an integer](https://github.com/nodejs/node/blob/5fc9d3c480bb23cd348169cd0f3c3d5c1799a287/lib/_http_server.js#L295) when the response is sent. So that means if `statusCode` is a stringified number then it'll be properly sent. I think maybe it makes more sense for us to do this ourselves though so that we can just send 500 if casting to a number doesn't work. I'll open another PR.

The other thing that happens in Node.js internals is that if the status code is invalid an [error is thrown](https://github.com/nodejs/node/blob/5fc9d3c480bb23cd348169cd0f3c3d5c1799a287/lib/_http_server.js#L297). This isn't ideal so I think we should also make sure that the status is between `400` and `599`. Again will PR this.